torch/fx/node.py

```diff
@@ -921,9 +921,7 @@ def map_aggregate(a: ArgumentT, fn: Callable[[Argument], Argument]) -> ArgumentT
     elif isinstance(a, list):
         result = immutable_list([map_aggregate(elem, fn) for elem in a])
     elif isinstance(a, dict):
-        result = immutable_dict()
-        for k, v in a.items():
-            dict.__setitem__(result, k, map_aggregate(v, fn))  # type: ignore[index]
+        return immutable_dict([(k, map_aggregate(v, fn)) for k, v in a.items()])  # type: ignore[return-value]
```

Comment:
Either benchmark this or let's keep the old version.  This was a performance optimization.

Reply:
Split to a separate PR.